sw/host/opentitantool/src/command/spi.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::fs::{self, File};
+use std::io::{self, Write};
+use std::time::Instant;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::io::spi::Transfer;
+use opentitanlib::spiflash::SpiFlash;
+use opentitanlib::transport::{Capability, Transport};
+
+/// Read and parse an SFDP table.
+#[derive(Debug, StructOpt)]
+pub struct SpiSfdp {
+    #[structopt(
+        short,
+        long,
+        help = "Display raw SFDP bytes rather than the parsed struct."
+    )]
+    raw: Option<usize>,
+}
+
+// Print a hexdump of a buffer to stdout.
+// The hexdump includes the offset, hex bytes and printable ASCII characters.
+//
+//  00000000: 53 46 44 50 06 01 02 ff 00 06 01 10 30 00 00 ff  SFDP........0...
+//  00000010: c2 00 01 04 10 01 00 ff 84 00 01 02 c0 00 00 ff  ................
+//  00000020: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
+//  00000030: e5 20 fb ff ff ff ff 3f 44 eb 08 6b 08 3b 04 bb  . .....?D..k.;..
+//
+// Note: This format can be consumed by `xxd -r` and converted back into binary.
+fn hexdump(buf: &[u8]) {
+    let mut ascii = [b'.'; 16];
+    for (i, v) in buf.iter().enumerate() {
```

Comment:
Maybe it makes sense to use `buf.chunks(16)` here, too? I hate to be like "nonono write this code" but I think you may find this bit of rust-foo none-the-less helpful:

```rust
for (i, word) in buf.chunks(16).enumerate() {
  let ascii = [b'.'; 16];
  print!("{:08x}:", i * 16);
  for (j, b) in word.iter().copied().enumerate() {
    if b == b' ' || b.is_ascii_graphic() { ascii[j] = b; }
    print!(" {:02x}", b);
  }
  for _ in 0..(16-i) {
    println!("   ");
  }
  println!("  {}", std::from_utf8(&ascii).unwrap());
}
```

Thoughts?

Reply:
Its shorter and I _do_ like that.  I'm somewhat ambivalent about the chunking -- I'm going to run through the entire array^W slice anyway.

The original version I wrote was a straight translation of what I have memorized for a C hexdump function.